packages/addon-dev/sample-typescript-rollup.config.mjs

```diff
@@ -0,0 +1,87 @@
+import ts from 'rollup-plugin-ts';
+import { defineConfig } from 'rollup';
+
+import { Addon } from '@embroider/addon-dev/rollup';
+
+const addon = new Addon({
+  srcDir: 'src',
+  destDir: 'dist',
+});
+
+export default defineConfig({
+  // https://github.com/rollup/rollup/issues/1828
+  watch: {
+    chokidar: {
+      usePolling: true,
+    },
+  },
+  output: {
+    ...addon.output(),
+    sourcemap: true,
+    // Until a bug with the ember-cli-htmlbars-removal transform is fixed, you
+    // may need
+    // hoistTransitiveImports: false
+    // specified here as well
+  },
+  plugins: [
+    // These are the modules that users should be able to import from your
+    // addon. Anything not listed here may get optimized away.
+    addon.publicEntrypoints([
+      'index.ts',
+      'components/**/*.ts',
+      'helpers/**/*.ts',
+      'modifiers/**/*.ts',
+      'services/**/*.ts',
+      'test-support/**/*.ts',
+    ]),
+
+    // These are the modules that should get reexported into the traditional
+    // "app" tree. Things in here should also be in publicEntrypoints above, but
+    // not everything in publicEntrypoints necessarily needs to go here.
+    addon.appReexports([
+      'index.js',
+      'components/**/*.js',
+      'helpers/**/*.js',
+      'modifiers/**/*.js',
+      'services/**/*.js',
+      'test-support/**/*.js',
+    ]),
+    // This babel config should *not* apply presets or compile away ES modules.
+    // It exists only to provide development niceties for you, like automatic
+    // template colocation.
+    // See `babel.config.json` for the actual Babel configuration!
+    ts({
+      // can be changed to swc or other transpilers later
+      // but we need the ember plugins converted first
+      // (template compilation and co-location)
+      transpiler: 'babel',
+      browserslist: ['last 2 firefox versions', 'last 2 chrome versions'],
```

Comment:
Why do you have this? Browser support should be an app concern.

Reply:
Yes, the docs confirm it: https://github.com/wessberg/rollup-plugin-ts#babelpreset-env-behavior-and-how-to-opt-out

Reply:
setting browserslist to false added poplyfills for ownKeys and objectSpread in my addon.

Reply:
@ef4 @NullVoxPopuli  As discussed yesterday, I looked into this today. Our suspicion, that `rollup-plugin-ts` does some unexpected magic, seems to be right unfortunately. I filed this issue: https://github.com/wessberg/rollup-plugin-ts/issues/189